### PR TITLE
remove prims::square

### PR DESCRIPTION
**Pull request by zasdfgbnm**
because it is just `x * x`

### CODE REVIEW

**Comment by ngimel** [comment]
@0x00b1 we can always have a ref for that then.

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -628,10 +628,15 @@ def _sigmoid(a: TensorLikeType) -> TensorLikeType:
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
 )
 
+
+def _square(a: TensorLikeType) -> TensorLikeType:
```
Comment (mruberry) [inline torch/_refs/__init__.py:632]: Nice!

File: torch/_refs/__init__.py [hunk 2]
```diff
@@ -628,10 +628,15 @@ def _sigmoid(a: TensorLikeType) -> TensorLikeType:
...
+    return mul(a, a)
+
+
 square = _make_elementwise_unary_reference(
-    prims.square,
+    _square,
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.BOOL_TO_LONG,
-    aten_op=None,  # CompositeImplicitAutograd,
+    aten_op=torch.ops.aten.square,
```
Comment (mruberry) [inline torch/_refs/__init__.py:639]: Nice

**Comment by mruberry** [comment]
I'm not totally sure what's going on with the tests failures -- @ezyang?

File: torch/_refs/__init__.py [hunk 3]
```diff
@@ -628,10 +628,15 @@ def _sigmoid(a: TensorLikeType) -> TensorLikeType:
...
 
+
+def _square(a: TensorLikeType) -> TensorLikeType:
+    return mul(a, a)
+
+
 square = _make_elementwise_unary_reference(
-    prims.square,
+    _square,
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.BOOL_TO_LONG,
```
[the remark this answers is not in the document]
Comment (zasdfgbnm) [reply torch/_refs/__init__.py:638]: But eager mode does not match with numpy:
```
a = numpy.array(True)
(numpy.square(a)).dtype
```
gives int8

File: torch/_refs/__init__.py (hunk 2 above)
Comment (ezyang) [reply torch/_refs/__init__.py:639]: don't do this, this is why the tests are failing

Comment (zasdfgbnm) [reply torch/_refs/__init__.py:639]: Thanks!

File: torch/_refs/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mruberry) [reply torch/_refs/__init__.py:638]: Which is also crazy!

**Review by mruberry — APPROVED** [review]
Cool! Thanks @zasdfgbnm

**Pull request closed without merging**